File: sorting-searching/sorting-algorithms/Branchless-sorting/branchless-sorting.js
function branchlessCompareSwap(a, b) {
  const minValue = Math.min(a, b);
  const maxValue = Math.max(a, b);

  return [minValue, maxValue];
}

function branchlessSort(arr) {
  const result = [...arr];

  for (let i = 0; i < result.length - 1; i++) {
    for (let j = 0; j < result.length - i - 1; j++) {
      [result[j], result[j + 1]] =
        branchlessCompareSwap(result[j], result[j + 1]);
    }
  }

  return result;
}

const arr = [8, 3, 7, 4, 9, 2, 6, 1];

console.log("Original array:", arr);

const sortedArray = branchlessSort(arr);

console.log("Sorted array:", sortedArray);

File: sorting-searching/sorting-algorithms/Branchless-sorting/branchless-sorting.py
def branchless_compare_swap(a, b):
    min_value = min(a, b)
    max_value = max(a, b)

    return min_value, max_value


def branchless_sort(arr):
    result = arr[:]

    n = len(result)

    for i in range(n - 1):
        for j in range(n - i - 1):
            result[j], result[j + 1] = branchless_compare_swap(
                result[j],
                result[j + 1]
            )

    return result


arr = [8, 3, 7, 4, 9, 2, 6, 1]

print("Original array:", arr)

sorted_array = branchless_sort(arr)

print("Sorted array:", sorted_array)
